bug: cannot get specific apibinding from `kcp claims get`

### Describe the bug

## `kcp` version

```                                                                                                                                                                          
kcp version v1.33.3+kcp-v0.29.1
```

## Issue

As part of my work on #3910, I've discovered that it seems impossible to fetch a specific APIBinding using `kcp claims get apibinding` resulting in panic.

```
nui@Q6VYXY6QK6 (feat/3910*) » k apply -f wildwest-schemas/apibinding.yaml
apibinding.apis.kcp.io/cowboys configured

nui@Q6VYXY6QK6 (feat/3910*) » k kcp claims get apibinding
APIBINDING   GROUP-RESOURCE   VERBS   STATUS
cowboys      configmaps       *       Rejected

nui@Q6VYXY6QK6 (feat/3910*) » k kcp claims get apibinding cowboys
panic: Unsupported APIBinding version provided.

goroutine 1 [running]:
github.com/kcp-dev/cli/pkg/helpers/apis/apis.NewAPIBindingList(...)
        /home/runner/work/kcp/kcp/staging/src/github.com/kcp-dev/cli/pkg/helpers/apis/apis/apibinding.go:86
github.com/kcp-dev/cli/pkg/claims/plugin.(*GetAPIBindingOptions).Run(0x14000518c80, {0x101ff0988, 0x102f051a0})
        /home/runner/work/kcp/kcp/staging/src/github.com/kcp-dev/cli/pkg/claims/plugin/claims.go:121 +0x530
github.com/kcp-dev/cli/pkg/claims/cmd.New.func3(0x140004a4f08, {0x14000502d40?, 0x4?, 0x101948a1f?})
        /home/runner/work/kcp/kcp/staging/src/github.com/kcp-dev/cli/pkg/claims/cmd/cmd.go:81 +0x58
github.com/spf13/cobra.(*Command).execute(0x140004a4f08, {0x14000502d00, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0x844
github.com/spf13/cobra.(*Command).ExecuteC(0x1400003c608)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(0x0?)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071 +0x1c
main.main()
        /home/runner/work/kcp/kcp/staging/src/github.com/kcp-dev/cli/cmd/kubectl-kcp/main.go:33 +0x88
```

It seems that [`NewAPIBindingList`](https://github.com/kcp-dev/kcp/blob/main/staging/src/github.com/kcp-dev/cli/pkg/helpers/apis/apis/apibinding.go#L78-L87) is expecting `*apisv1alpha2.APIBinding` type but receives [`*apis.apiBindingV1alpha2`]() type instead, causing the `panic`.

### Steps To Reproduce

0. Follow [quickstart tutorial](https://docs.kcp.io/kcp/v0.30/concepts/quickstart-tenancy-and-apis/#use-those-apis-as-a-service-consumer)
1. Create `cowboys` APIBinding

```
k apply -f <<EOF
apiVersion: apis.kcp.io/v1alpha2
kind: APIBinding
metadata:
  name: cowboys
spec:
  permissionClaims:
    - group: ""
      resource: configmaps
      selector:
        matchAll: true
      state: Rejected
      verbs:
        - '*'
  reference:
    export:
      name: wildwest.dev
      path: root:wildwest:cowboys-service
EOF
```
2. Run `kcp claims get apibinding` --> list out `cowboys`
3. Run `kcp claims get apibinding cowboys` --> panic

### Expected Behaviour

Run `kcp claims get apibinding cowboys` should return `cowboys` permission claims.

### Additional Context

_No response_

--- a/staging/src/github.com/kcp-dev/cli/pkg/helpers/apis/apis/apibinding.go
+++ b/staging/src/github.com/kcp-dev/cli/pkg/helpers/apis/apis/apibinding.go
@@ -77,11 +77,10 @@ func NewAPIBinding(nativeBinding any) APIBinding {
 
 func NewAPIBindingList(nativeBinding any) APIBindingList {
 	switch asserted := nativeBinding.(type) {
-	case *apisv1alpha1.APIBinding:
-		return &apiBindingListV1alpha1{bindings: []*apisv1alpha1.APIBinding{asserted}}
-	case *apisv1alpha2.APIBinding:
-		return &apiBindingListV1alpha2{bindings: []*apisv1alpha2.APIBinding{asserted}}
+	case *apiBindingV1alpha2:
+		return &apiBindingListV1alpha2{bindings: []*apisv1alpha2.APIBinding{asserted.binding}}
+	case *apiBindingV1alpha1:
+		return &apiBindingListV1alpha1{bindings: []*apisv1alpha1.APIBinding{asserted.binding}}
 	}
-
 	panic("Unsupported APIBinding version provided.")
 }

